BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
```

Comment:
We should introduce a class to hold all relevant request parameters to this service. With the current design, we have to pass the list of currencies as a bare json array in the request. As we are expected to add support for more source currencies down the line, this has to change. It's better to do it now, otherwise we would have to introduce a breaking API design change later, which is always good to avoid.

Let's create a request class. Note the use of `required` keyword on the property. This makes it so the class can't be instantiated without the property being initialized to a non-null value. This is generally what we want for simple data transfer objects like this.

```
public class ExchangeRatesRequest
{
    [JsonPropertyName("targetCurrencies")]
    public required List<string> TargetCurrencies { get; set; }
}
```

We then use this class as the parameter instead:

`public IEnumerable<Rate> GetExchangeRates(ExchangeRatesRequest request)`

... and use `request.TargetCurrencies` where we previously used `currencies`.